### PR TITLE
Update pure_powervc.py

### PR DESCRIPTION
**Pull request by Maria06**
Make available pure storage port details with  port status and speed

### CODE REVIEW

**Comment by Maria06** [comment]
@shyamav Please review the change set

File: pure_powervc.py [hunk 1]
```diff
@@ -30,9 +30,6 @@
 from cinder.volume import volume_types
 
 from powervc_cinder.volume import discovery_driver
-from powervc_cinder.volume.discovery_driver import PORT_LOCATION
-from powervc_cinder.volume.discovery_driver import PORT_STATUS
-from powervc_cinder.volume.discovery_driver import UNKNOWN_VALUE
```
Comment (Maria06) [inline pure_powervc.py]: Retain the constants

File: pure_powervc.py [hunk 2]
```diff
@@ -60,17 +59,26 @@ def get_vendor_str(self):
...
+            for count in range(0, len(hardware)):
+                if hardware[count]['name'] == port.get('name'):
+                    pinfo = {
+                        'wwpn': port.get('wwn'),
+                        'port_name': port.get('name'),
+                        PORT_LOCATION : UNKNOWN_VALUE,
+                        PORT_STATUS: (
+                            'online' if hardware[count]['status'] == 'ok'
+                            else hardware[count]['status']),
+                        'speed': hardware[count]['speed']}
```
Comment (shyamav) [inline pure_powervc.py]: We need to convert the speed to Gib

**Review by sdodsley — COMMENTED** [review]
I am OK with these changes.
If you want to update the code to get the FC speed in GiB, go ahead and do that, then I will approve and merge this.

File: pure_powervc.py [hunk 3]
```diff
@@ -60,17 +59,38 @@ def get_vendor_str(self):
...
-        return ports
+            self._add_fabric_mapping(available_ports)
+        return available_ports
+
+    def convert_fc_port_speed(self, speed):
+        if speed == 0:
+            return 0
+        else:
+            speed_byte = float(speed)
+            speed_kb = float(1000)
```
Comment (shyamav) [inline pure_powervc.py:90]: we can make this one line.

**Pull request merged**